OMCompiler/SimulationRuntime/c/simulation/solver/jacobianSymbolical.c

```diff
@@ -0,0 +1,77 @@
+/*
+ * This file is part of OpenModelica.
+ *
+ * Copyright (c) 1998-2016, Open Source Modelica Consortium (OSMC),
+ * c/o Linköpings universitet, Department of Computer and Information Science,
+ * SE-58183 Linköping, Sweden.
+ *
+ * All rights reserved.
+ *
+ * THIS PROGRAM IS PROVIDED UNDER THE TERMS OF THE BSD NEW LICENSE OR THE
+ * GPL VERSION 3 LICENSE OR THE OSMC PUBLIC LICENSE (OSMC-PL) VERSION 1.2.
+ * ANY USE, REPRODUCTION OR DISTRIBUTION OF THIS PROGRAM CONSTITUTES
+ * RECIPIENT'S ACCEPTANCE OF THE OSMC PUBLIC LICENSE OR THE GPL VERSION 3,
+ * ACCORDING TO RECIPIENTS CHOICE.
+ *
+ * The OpenModelica software and the OSMC (Open Source Modelica Consortium)
+ * Public License (OSMC-PL) are obtained from OSMC, either from the above
+ * address, from the URLs: http://www.openmodelica.org or
+ * http://www.ida.liu.se/projects/OpenModelica, and in the OpenModelica
+ * distribution. GNU version 3 is obtained from:
+ * http://www.gnu.org/copyleft/gpl.html. The New BSD License is obtained from:
+ * http://www.opensource.org/licenses/BSD-3-Clause.
+ *
+ * This program is distributed WITHOUT ANY WARRANTY; without even the implied
+ * warranty of MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE, EXCEPT AS
+ * EXPRESSLY SET FORTH IN THE BY RECIPIENT SELECTED SUBSIDIARY LICENSE
+ * CONDITIONS OF OSMC-PL.
+ *
+ */
+
+ /*! \file jacobian_symbolical.c
+ */
+
+#include "simulation/solver/jacobianSymbolical.h"
+
+
+void genericParallelColoredSymbolicJacobianEvaluation(int rows, int columns, SPARSE_PATTERN* spp,
+                                                      void* matrixA, ANALYTIC_JACOBIAN* jacColumns, DATA* data,
+                                                      threadData_t* threadData,
+                                                      void (*f)(int, int, int, double, void*, int))
+{
+  ANALYTIC_JACOBIAN* t_jac = jacColumns;
+
+  unsigned int ii, j, l, nth, i;
```

Comment:
A potentially dangerous declaration, ii is never used and i and j is redefined in the for loops. Consider removal of ii, j and i. Also "l" might not be the best name, consider adding a comment explaining the intention behind l or rename it :)

Reply:
Renamed it and updated the description of the function. Also removed a unused function.